timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -406,7 +417,9 @@ def slice_by_timestep(self, time_step_slice: slice) -> TimeSeriesDataFrame:
             idx = pd.MultiIndex.from_product([(ix,), data_slice.index], names=[ITEMID, TIMESTAMP])
             data_slice.set_index(idx, inplace=True)
             slices.append(data_slice)
-        return self.__class__(pd.concat(slices), static_features=self.static_features)
+        slice_df = self.__class__(pd.concat(slices), static_features=self.static_features)
+        slice_df._cached_freq = self._cached_freq
```

Comment:
We might need to recompute the `_cached_freq` here - for example, a user could specify `time_step_slice=slice(None, None, 2)` which would upsample the time series

Reply:
Hmm, right. But I think I will rather block this kind of upsampling here as it may cause other problems we don't know. What do you think?

Reply:
I thought the whole point of passing a `slice(...)` object here instead of something like `start_idx=None, end_idx=None` here was to allow upsampling 😅 

I'm okay with disabling subsampling. Should we change the API in this case to only allow start_idx / end_idx (in a separate PR)?

Reply:
I would take care of this in #2020, if we decide to go down this route